Lib/logging/__init__.py

```diff
@@ -231,25 +229,20 @@ def _checkLevel(level):
 #
 _lock = threading.RLock()
 
+
+@contextmanager
 def _acquireLock():
```

Comment:
Changing the usage of the existing API sounds like a bad idea. I suggest to add a new context manager API and leave thr existing API unchanged.

Reply:
Good point! I've moved the new logic to `_acquireModuleLock`, which utilises the other two methods